pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -0,0 +1,292 @@
+		check:              checkCrcNetworkManagerConfig,
+		fixDescription:     "Writing Network Manager config for crc",
+		fix:                fixCrcNetworkManagerConfig,
+		cleanupDescription: "Removing /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf file",
+		cleanup:            removeCrcNetworkManagerConfig,
+	},
+	{
+		configKeySuffix:    "check-crc-dnsmasq-file",
+		checkDescription:   "Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists",
+		check:              checkCrcDnsmasqConfigFile,
+		fixDescription:     "Writing dnsmasq config for crc",
+		fix:                fixCrcDnsmasqConfigFile,
+		cleanupDescription: "Removing /etc/NetworkManager/dnsmasq.d/crc.conf file",
+		cleanup:            removeCrcDnsmasqConfigFile,
+	},
+}
+
+const (
+	crcDnsmasqConfigFile        = "crc.conf"
+	crcNetworkManagerConfigFile = "crc-nm-dnsmasq.conf"
+	crcNMDispatcherFile         = "99-crc.sh"
+)
+
+var (
+	crcDnsmasqConfigPath = filepath.Join(string(filepath.Separator), "etc", "NetworkManager", "dnsmasq.d", crcDnsmasqConfigFile)
+	crcDnsmasqConfig     = `server=/apps-crc.testing/192.168.130.11
+server=/crc.testing/192.168.130.11
+`
+	crcNetworkManagerConfigPath = filepath.Join(string(filepath.Separator), "etc", "NetworkManager", "conf.d", crcNetworkManagerConfigFile)
+	crcNetworkManagerConfig     = `[main]
+dns=dnsmasq
+`
+	crcNMDispatcherPath   = filepath.Join(string(filepath.Separator), "etc", "NetworkManager", "dispatcher.d", "pre-up.d", crcNMDispatcherFile)
+	crcNMDispatcherConfig = `#!/bin/sh
+# This is a NetworkManager dispatcher script to configure split DNS for
+# the 'crc' libvirt network.
+# The corresponding crc bridge is recreated each time the system reboots, so
+# it cannot be configured permanently through NetworkManager.
+# Changing DNS settings with nmcli requires the connection to go down/up,
+# so we directly make the change using resolvectl
+
+export LC_ALL=C
+
+if [ "$1" = crc ]; then
+        resolvectl domain "$1" ~testing
+        resolvectl dns "$1" 192.168.130.11
+        resolvectl default-route "$1" false
+fi
+
+exit 0
+`
+)
+
+var systemdResolvedPreflightChecks = [...]Check{
+	{
+		configKeySuffix:  "check-systemd-resolved-running",
+		checkDescription: "Checking if the systemd-resolved service is running",
+		check:            checkSystemdResolvedIsRunning,
+		fixDescription:   "systemd-resolved is required on this distribution. Please make sure it is installed and running manually",
+		flags:            NoFix,
+	},
+	{
+		configKeySuffix:    "check-crc-nm-dispatcher-file",
+		checkDescription:   fmt.Sprintf("Checking if %s exists", crcNMDispatcherPath),
+		check:              checkCrcNMDispatcherFile,
+		fixDescription:     "Writing NetworkManager dispatcher file for crc",
+		fix:                fixCrcNMDispatcherFile,
+		cleanupDescription: fmt.Sprintf("Removing %s file", crcNMDispatcherPath),
+		cleanup:            removeCrcNMDispatcherFile,
+	},
+}
+
+func checkNMConfigFile(path string, expectedContent string) error {
```

Comment:
Generic file-exists-and-matches-content check, might be better of in a different pkg.

Reply:
Good point, fits well with https://github.com/code-ready/crc/blob/cc9db2fcceaf8c94a64f865e8a1202d27aaf9513/pkg/os/util.go#L64